ApollosProject/apollos-prototype
# Clears apollo store instead of reseting store

## DESCRIPTION

### What does this PR do, or why is it needed?

There's a problem in the current app where any action that triggers a store reset (logout or auth error) causes all of the queries to be rerun. This can cause a vast number of queries to be rerun in some cases, as evidenced by watching your API console after logging out of the app (even after not clicking on a single content item or piece of UI besides the logout button!) 

This causes problems for two reasons: 

1. These queries take time and energy to be ran and re-ran
2. Some of those queries aren't designed to be run without a user active, and return auth errors. 

We have a workaround to prevent infinite redirects because of the second issue, but Bugsnag is still hard to sift through at least in part because of the number of unnecessary errors. 

By switching to `client.clearStore` and `client.onClearStore` we can wipe the cache without reloading the data. You can see that in action on the attached GIF of logging out. 

I needed to update apollo-client, because our version of `onClearStore` would fire the callback _before_ the store had been wiped, meaning the data we would write with `onClearStore` would immediately be erased. 

### What design trade-offs/decisions were made?

### How do I test this PR?

1. Login to the app.
2. Logout of the app.
  - Clear the API console before you do so, ensure that only 1 or 2 queries are fired.
3. Log into the app.
4. Play a media player video 
  - In the past, messing with this functionality has caused the media player to stop working. 

### What automated tests did you write?

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
